File: core/src/main/java/com/egoal/darkestpixeldungeon/items/weapon/melee/SorceressWand.java
package com.egoal.darkestpixeldungeon.items.weapon.melee;

import com.egoal.darkestpixeldungeon.items.weapon.enchantments.Unstable;
import com.egoal.darkestpixeldungeon.sprites.ItemSprite;
import com.egoal.darkestpixeldungeon.sprites.ItemSpriteSheet;
import com.watabou.noosa.Image;

/**
 * Created by 93942 on 4/23/2018.
 */

public class SorceressWand extends MeleeWeapon {

  {
    image = ItemSpriteSheet.DPD_SORCERESS_WAND;
    tier = 1;
    DLY = 1.f;
    unique = true;

  }

  public SorceressWand() {
    // give enchantment
    enchant(new Unstable());
  }

  // 1~7
  @Override
  public int max(int lvl) {
    return 4 * tier + 3 +    //base
            lvl * (tier + 1);   //level scaling
  }

  @Override
  public int STRReq(int lvl) {
    lvl = Math.max(0, lvl);
    //strength req decreases at +1,+3,+6,+10,etc.
    return (7 + tier * 2) - (int) (Math.sqrt(8 * lvl + 1) - 1) / 2;
  }
}

File: core/src/main/java/com/egoal/darkestpixeldungeon/items/weapon/melee/SorceressWand.kt
package com.egoal.darkestpixeldungeon.items.weapon.melee

import com.egoal.darkestpixeldungeon.items.weapon.enchantments.Unstable
import com.egoal.darkestpixeldungeon.sprites.ItemSprite
import com.egoal.darkestpixeldungeon.sprites.ItemSpriteSheet
import com.watabou.noosa.Image

/**
 * Created by 93942 on 4/23/2018.
 */

class SorceressWand : MeleeWeapon() {

    init {
        image = ItemSpriteSheet.DPD_SORCERESS_WAND
        tier = 1
        DLY = 1f
        unique = true

        // give enchantment
        enchant(Unstable::class.java, 30)
    }

    override fun min(lvl: Int): Int = super.min(lvl) + 1

    // 1~7
    override fun max(lvl: Int): Int = 4 * tier + 3 + lvl * (tier + 1)

    override fun STRReq(lvl: Int): Int = super.STRReq(lvl) - 1
}
